tests/models/whisper/test_modeling_whisper.py

```diff
@@ -1990,7 +1787,7 @@ def test_tiny_generation(self):
         generated_ids = model.generate(input_features, num_beams=5, max_length=20)
         transcript = processor.tokenizer.decode(generated_ids[0])
 
-        EXPECTED_TRANSCRIPT = " Mr. Quilter is the apostle of the middle classes and we are glad to welcome his gospel"
+        EXPECTED_TRANSCRIPT = "<|startoftranscript|><|en|><|transcribe|><|notimestamps|> Mr. Quilter is the apostle of the middle classes and we are glad to welcome his gospel"
```

Comment:
As discussed offline, I don't think we should have behavior that depends on an internal call over which the user has no influence.

Before, we had those extra tokens if the audio was short (<30s). Now, we add to this an internal condition: there's only one call to the internal `generate`.

IMO, we should have a consistent behaviour: removing these special tokens by default. And add an additional flag to allow this edge case.

Reply:
fixed in [0ea240f](https://github.com/huggingface/transformers/pull/34135/commits/0ea240fa88d2c7aeb3b8968427fb3f15a8e76e15) 👌